pkg/db/src/implementations/version-1.2/books.ts

```diff
@@ -37,10 +37,29 @@ class Books implements BooksInterface {
 	}
 
 	async get(isbns: string[]): Promise<(BookEntry | undefined)[]> {
-		const rawBooks = await this.#db._pouch.allDocs<BookEntry>({ keys: isbns.map((isbn) => `books/${isbn}`), include_docs: true });
+		const rawBooks = await this.#db._pouch
+			.allDocs<BookEntry>({ keys: isbns.map((isbn) => `books/${isbn}`), include_docs: true })
+			.then((docs) => unwrapDocs(docs));
 		// The rows are returned in the same order as the supplied keys array.
 		// The row for a nonexistent document will just contain an "error" property with the value "not_found".
-		return unwrapDocs(rawBooks);
+
+		const combinedResults = wrapIter(isbns).zip(rawBooks);
+
+		const existingBooks: BookEntry[] = [];
+		// temp fix for isbn
+		const fetchIsbns = combinedResults.map(([isbn, book]) => {
+			if (book !== undefined) {
+				existingBooks.push(book);
+				return;
+			}
+
+			return isbn;
+		});
+
+		const fetchedBooks = await Promise.all(fetchIsbns.map((isbn = "") => pluginManager.getBook(isbn)));
+		this.upsert(fetchedBooks);
+
+		return [...fetchedBooks, ...existingBooks];
```

Comment:
Ok, this messes up the order and is unnecessary. What we want to do is still return the full data frame (`rawBooks` in this implementation).

The fetching of not-found books is done in the background and, when fetched, it will trigger another pass through the stream